unit/solvers/strings/string_constraint_generator_main/char_set_restrictions.cpp

```diff
@@ -0,0 +1,33 @@
+/*******************************************************************\
+
+Module: Unit tests for char_set_restrictions in
+        solvers/refinement/string_constraint_generator_main.cpp
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include <testing-utils/use_catch.h>
+
+#include <solvers/strings/string_constraint_generator.h>
+#include <util/std_types.h>
+
+TEST_CASE(
+  "char_set_restrictions",
+  "[core][solvers][strings][string_constraint_generator_main]")
+{
+  const std::string char_set = "a-z";
+  const exprt expr = exprt("dummy", unsignedbv_typet(16));
+  const auto result = char_set_restrictions(expr, char_set);
+  REQUIRE(can_cast_expr<and_exprt>(result));
+
+  REQUIRE(result.op0().id() == ID_ge);
+  REQUIRE(result.op0().op0().type() == unsignedbv_typet(16));
+  REQUIRE(can_cast_expr<constant_exprt>(result.op0().op1()));
+  REQUIRE(to_constant_expr(result.op0().op1()).get_value() == "61");
+
+  REQUIRE(result.op1().id() == ID_le);
+  REQUIRE(result.op1().op0().type() == unsignedbv_typet(16));
+  REQUIRE(can_cast_expr<constant_exprt>(result.op1().op1()));
+  REQUIRE(to_constant_expr(result.op1().op1()).get_value() == "7A");
```

Comment:
It's hard to see what you're trying to assert here. Suggest documenting with (`GIVEN` `WHEN` `THEN`).